dev/release/README.md

```diff
@@ -0,0 +1,210 @@
+```
+
+Here is an example to release 21.0.0 RC0:
+
+```console
+$ dev/release/release_rc.sh 21.0.0 0
+```
+
+The arguments of `release_rc.sh` are the version and the RC number. If
+RC0 has a problem, we'll increment the RC number such as RC1, RC2 and
+so on.
+
+## Publish
+
+We need to do the followings to publish a new release:
+
+* Publish the source archive to apache.org
+
+Run `dev/release/release.sh` on a working copy of
+`git@github.com:apache/arrow-swift` not your fork to publish the
+source archive to apache.org:
+
+```console
+$ dev/release/release.sh ${VERSION} ${RC}
+```
+
+Here is an example to release 21.0.0 RC0:
+
+```console
+$ dev/release/release.sh 21.0.0 0
+```
+
+Add the release to ASF's report database via [Apache Committee Report
+Helper](https://reporter.apache.org/addrelease.html?arrow).
+
+## Announce the new release on the mailing list
+
+Send an email to "announce@apache.org" from your Apache email, CC'ing
+dev@arrow.apache.org/user@arrow.apache.org.  See an [example
+post](https://lists.apache.org/thread/bxpt0r8kw0ltgywnylqdroskkt6966z4).
+
+```
+To: announce@apache.org
+CC: dev@arrow.apache.org, user@arrow.apache.org
+Subject: [ANNOUNCE] Apache Arrow Swift 21.0.0 released
+
+The Apache Arrow community is pleased to announce the Arrow Swift 21.0.0
+release.
+
+The release is available now.
+
+Source archive:
+  https://www.apache.org/dyn/closer.lua/arrow/apache-arrow-swift-21.0.0/
+
+On Swift Package Index:
+  https://swiftpackageindex.com/apache/arrow
+
+Read the full changelog:
+  https://github.com/apache/arrow-swift/releases/tag/v21.0.0
+
+What is Apache Arrow?
+---------------------
+
+Apache Arrow is a universal columnar format and multi-language toolbox
+for fast data interchange and in-memory analytics. It houses a set of
+canonical in-memory representations of flat and hierarchical data
+along with multiple language-bindings for structure manipulation. It
+also provides low-overhead streaming and batch messaging, zero-copy
+interprocess communication (IPC), and vectorized in-memory analytics
+libraries.
+
+Please report any feedback to the GitHub repository:
+  https://github.com/apache/arrow-swift/issues
+  https://github.com/apache/arrow-swift/discussions
+
+Regards,
+The Apache Arrow community.
+```
+
+## Announce the new release on social media
+
+Make a post on our [BlueSky](https://bsky.app/profile/arrow.apache.org) and
+[LinkedIn](https://www.linkedin.com/company/apache-arrow/) accounts. (Ask
+your fellow PMC members for access if need be, or ask a PMC member to make the
+post on your behalf.)  The post should link to the blog post. See [example
+BlueSky post](https://bsky.app/profile/arrow.apache.org/post/3lioi6ov5h22d)
+and [example LinkedIn post](https://www.linkedin.com/posts/apache-arrow_apache-arrow-java-1820-release-activity-7298633716522758144-L71x).
+
+## Verify
+
+We have a script to verify a RC.
+
+You must install the following commands to use the script:
+
+* `curl`
```

Comment:
Should we note that go and swift are also required to run the script? We can also update the `verify_rc.sh` script on a follow up PR to download the latest versions of go and swift if not present.